src/shackle_pool.erl

```diff
@@ -215,32 +291,18 @@ server_mod(shackle_tcp) ->
 server_mod(shackle_udp) ->
     shackle_udp_server.
 
-server_spec(ServerMod, ServerName, Name, Client, ClientOptions, Index) ->
-    StartFunc = {ServerMod, start_link,
-        [ServerName, Name, Client, ClientOptions, Index]},
-    {ServerName, StartFunc, permanent, 5000, worker, [ServerMod]}.
 
-start_children(Name, Client, ClientOptions, #pool_options{
-    pool_size = PoolSize
-}) ->
-    
+start_workers(Name, Client, ClientOptions, #pool_options{pool_size = PoolSize}) ->
     Protocol = ?LOOKUP(protocol, ClientOptions, ?DEFAULT_PROTOCOL),
     ServerMod = server_mod(Protocol),
-    ServerSpecs = [
+    [
         begin
             ServerName = server_name(Name, N),
-            server_spec(ServerMod, ServerName, Name,
-                Client, ClientOptions, N)
-        end || N <- lists:seq(1, PoolSize)],
-    [supervisor:start_child(?SUPERVISOR, ServerSpec) ||
-        ServerSpec <- ServerSpecs].
-
-stop_children([]) ->
-    ok;
-stop_children([ServerName | T]) ->
-    supervisor:terminate_child(?SUPERVISOR, ServerName),
-    supervisor:delete_child(?SUPERVISOR, ServerName),
-    stop_children(T).
+            {ok, Pid} = ServerMod:start_link(ServerName, Name, Client, ClientOptions, N),
+            {Pid, #{server_mod => ServerMod, serveer_name => ServerName,
+                pool_name => Name, client=> Client, client_options => ClientOptions, index => N}}
```

Comment:
As we're using gen_server here and use a map as the server state, just wondering if 'index' is really needed any more?

Reply:
yes it is needed, when dispatching traffic from marina to shackle, it will use the configured Stretegy to pick up a worker, we will need index there.